pkg/provider/ibm/provider.go

```diff
@@ -580,28 +582,43 @@ func (ibm *providerIBM) ValidateStore(store esv1beta1.GenericStore) error {
 	}
 
 	containerRef := ibmSpec.Auth.ContainerAuth
-	secretKeyRef := ibmSpec.Auth.SecretRef.SecretAPIKey
-	if utils.IsNil(containerRef.Profile) || (containerRef.Profile == "") {
-		// proceed with API Key Auth validation
-		err := utils.ValidateSecretSelector(store, secretKeyRef)
-		if err != nil {
-			return err
-		}
-		if secretKeyRef.Name == "" {
-			return fmt.Errorf("secretAPIKey.name cannot be empty")
-		}
-		if secretKeyRef.Key == "" {
-			return fmt.Errorf("secretAPIKey.key cannot be empty")
+	secretRef := ibmSpec.Auth.SecretRef
+
+	missingContainerRef := utils.IsNil(containerRef)
+	missingSecretRef := utils.IsNil(secretRef)
+
+	if missingContainerRef == missingSecretRef {
+		// since both are equal, if one is missing assume both are missing
+		if missingContainerRef {
+			return fmt.Errorf("missing auth method")
 		}
-	} else {
+		return fmt.Errorf("too many auth methods defined")
+	}
+
+	if !missingContainerRef && containerRef.Profile != "" {
```

Comment:
if profile is empty, we would not get into this branch, hence accessing `secretRef.SecretAPIKey` below which will be a nil pointer dereference panic.

Reply:
Ah yup, I see what you mean. I've updated the check for the profile to be nested inside the `!missingContainerRef` one and to return an error